appenders/python/pulse/formatters.py

```diff
@@ -26,5 +29,13 @@ def format(self, record):
         data["level"] = record.levelname
         data["message"] = record.msg
         data["threadName"] = record.threadName
+        while not 'hostname' in data: 
```

Comment:
Why is this a while loop? Could this get stuck? I don't think we want to retry, and also looking closer we don't want to do this for every method call for efficiency reasons. It should happen once at the class level and store it in a field.

Reply:
`PULSE_DEBUG = os.getenv('PULSE_DEBUG', False)` This can go at the class level also

Reply:
`If we log the message we could end up in an infinite loop.` I misunderstood this words.